test/integration/apiserver/cel/validatingadmissionpolicy_test.go

```diff
@@ -0,0 +1,942 @@
+					Name: "test-k8s",
+				},
+			},
+			err: "",
+		},
+		{
+			name: "runtime error when validating namespace, but failurePolicy=Fail",
+			policy: &admissionregistrationv1alpha1.ValidatingAdmissionPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "validate-namespace-suffix",
+				},
+				Spec: admissionregistrationv1alpha1.ValidatingAdmissionPolicySpec{
+					MatchConstraints: &admissionregistrationv1alpha1.MatchResources{
+						ResourceRules: []admissionregistrationv1alpha1.NamedRuleWithOperations{
+							{
+								RuleWithOperations: admissionregistrationv1alpha1.RuleWithOperations{
+									Operations: []admissionregistrationv1.OperationType{
+										"CREATE",
+									},
+									Rule: admissionregistrationv1.Rule{
+										APIGroups: []string{
+											"",
+										},
+										APIVersions: []string{
+											"*",
+										},
+										Resources: []string{
+											"namespaces",
+										},
+									},
+								},
+							},
+						},
+					},
+					Validations: []admissionregistrationv1alpha1.Validation{
+						{
+							Expression: "object.nonExistentProperty == 'someval'",
+						},
+					},
+					FailurePolicy: &failurePolicy,
+				},
+			},
+			policyBinding: &admissionregistrationv1alpha1.ValidatingAdmissionPolicyBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "validate-namespace-suffix-binding",
+				},
+				Spec: admissionregistrationv1alpha1.ValidatingAdmissionPolicyBindingSpec{
+					PolicyName: "validate-namespace-suffix",
+				},
+			},
+			namespace: &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-k8s",
+				},
+			},
+			err:           "ValidatingAdmissionPolicy 'validate-namespace-suffix' with binding 'validate-namespace-suffix-binding' denied request: expression 'object.nonExistentProperty == 'someval'' resulted in error: no such key: nonExistentProperty",
+			failureReason: metav1.StatusReasonInvalid,
+		},
+	}
+
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.CELValidatingAdmission, true)()
+			server, err := apiservertesting.StartTestServer(t, nil, []string{
+				"--enable-admission-plugins", "ValidatingAdmissionPolicy",
+			}, framework.SharedEtcd())
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer server.TearDownFn()
+
+			config := server.ClientConfig
+
+			client, err := clientset.NewForConfig(config)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), testcase.policy, metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicyBindings().Create(context.TODO(), testcase.policyBinding, metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			// TODO: add retry logic instead
+			time.Sleep(time.Second)
```

Comment:
Can `wait.PollImmediate()` be used here? E.g.

https://github.com/kubernetes/kubernetes/blob/5d06c6a92dfa8851348468d2bafbe787d6cb1229/test/integration/apiserver/admissionwebhook/reinvocation_test.go#L396

Reply:
Note that if the goals it to poll for a failed CREATE, then make sure that CREATEs that are accidentally successful needs to be considered don't cause problems.

Reply:
Fixed